kong/plugins/oidc/handler.lua

```diff
@@ -29,25 +30,44 @@ end
 
 function handle(oidcConfig, oidcSessionConfig)
   local response
+
+  -- attempt introspection of potential bearer token
   if oidcConfig.introspection_endpoint then
     response = introspect(oidcConfig)
+    -- if response, then introspect successful
     if response then
-      utils.injectUser(response)
+      local access_token = utils.get_bearer_access_token_from_header(oidcConfig)
+      local user_info, err = get_userinfo(oidcConfig, response)
+      
+      -- introspect passed but userinfo call failed, maybe we have id_token
```

Comment:
@csmcgee update the comment
introspect passed but userinfo call failed, instead of leaving userinfo blank. attach introspected token information

Reply:
this can fail if id_token is used for bearer token instead of access token.